command_test.go

```diff
@@ -401,3 +407,57 @@ func TestCancel(t *testing.T) {
 	}
 	<-wait
 }
+
+type testEmitterWithError testing.T
+
+func (s *testEmitterWithError) Close() error {
+	return nil
+}
+
+func (s *testEmitterWithError) SetLength(_ uint64) {}
+
+func (s *testEmitterWithError) SetError(err interface{}, code cmdkit.ErrorType) {
+	errorCount++
```

Comment:
We *definitely* shouldn't be using a global here. Instead, I'd stash the error in the test emitter (or set a flag or something).